lib/flight_job/command.rb

```diff
@@ -73,10 +75,59 @@ def pastel
       @pastel ||= Pastel.new
     end
 
+    Pager = Struct.new(:retry_file, :follow, :pastel) do
+      def page(arg)
+        path = arg.is_a?(Hash) ? arg[:path] : arg.to_s
+        if path
+          open_path(path) { |io| page_io(io) }
+        else
+          io = StringIO.new(arg.to_s)
```

Comment:
This still doesn't make any sense to me.  As far as I can see `arg` can be one of three things

1. a Hash with a `:path` key with a truthy value
2. a Hash without a `:path` key or with a `:path` key with a falsy value
3. not a Hash

Only for case (2) do we take the `StringIO` route.  But I'm not sure why we're calling `to_s` on a Hash.

Am I missing something here?

Reply:
I agree the method signature is odd, but the `page` method is a drop in replacement for `TTY::Pager`. It needs to support two use cases:

* `pager.page(file: ...)`
* `pager.page("string")`

In general I dislike this type of signature as it is awkward to implement. However I want to avoid differing from `TTY:Pager` too much, as this will make retro-fitting other apps easier (if an when required).

---

As far as the `edge` case where `arg` is a `Hash` but `path` is falsey, _do not do it_. This doesn't appear in the code base and isn't supported.

If you prefer, I can modify the method signature to be:

```
def page(str = nil, path: nil)
end
```

However this uses an optional positional argument, which I also dislike.

Reply:
Updated the method signature, it should be _vaguely_ sane now:

679ef321242b9defc8e3b2b5cf0e161e42f12825